website/docs/configuration/expressions.html.md

```diff
@@ -16,195 +16,204 @@ references to data exported by resources, arithmetic, conditional evaluation,
 * `bool`: either `true` or `false`. `bool` values can be used in conditional
   logic.
+* `list` (or `tuple`): a sequence of values, like
+  `["us-west-1a", "us-west-1c"]`. Elements in a list or tuple are identified by
+  consecutive whole numbers, starting with zero.
+* `map` (or `object`): a group of values identified by named labels, like
+  `{name = "Mabel", age = 52}`.
 
-The Terraform language will automatically convert `number` and `bool` values
-to `string` values when needed, and vice-versa as long as the string contains
-a valid representation of a number of boolean value.
-
-* `true` converts to `"true"`, and vice-versa
-* `false` converts to `"false"`, and vice-versa
-* `15` converts to `"15"`, and vice-versa
-
-### Collection Types
+Strings, numbers, and bools are sometimes called _primitive types._ Lists/tuples and maps/objects are sometimes called _complex types,_ _structural types,_ or _collection types._
 
-A _collection_ type allows multiple values of another type to be grouped
-together as a single value. The type of value _within_ a collection is called
-its _element type_, and all collection types must have an element type.
+Finally, there is one special value that has _no_ type:
 
-For example, the type `list(string)` means "list of strings", which is a
-different type than `list(number)`, a list of numbers. All elements of a
-collection must always be of the same type.
+* `null`: a value that represents _absence_ or _omission._ If you set an
+  argument of a resource or module to `null`, Terraform behaves as though you
+  had completely omitted it — it will use the argument's default value if it has
+  one, or raise an error if the argument is mandatory. `null` is most useful in
+  conditional expressions, so you can dynamically omit an argument if a
+  condition isn't met.
 
-The three _collection type kinds_ in the Terraform language are:
+### Advanced Type Details
 
-* `list(...)`: a sequence of values identified by consecutive whole numbers
-  starting with zero.
-* `map(...)`: a collection of values where each is identified by a string label.
-* `set(...)`: a collection of unique values that do not have any secondary
-  identifiers or ordering.
+In most situations, lists and tuples behave identically, as do maps and objects.
+Whenever the distinction isn't relevant, the Terraform documentation uses each
+pair of terms interchangeably (with a historical preference for "list" and
+"map").
 
-There is no direct syntax for creating collection type values, but the
-Terraform language can automatically convert a structural type value (as
-defined in the next section) to a similar collection type as long as all
-of its elements can be converted to the required element type.
+However, module authors and provider developers should understand the
+differences between these similar types (and the related `set` type), since they
+offer different ways to restrict the allowed values for input variables and
+resource arguments.
```

Comment:
And this is the spot where I nudge you to go learn the differences, if you're one of the people who actually needs to know. (There's a link on the next line, but the diff interleaved a big block of removed text here, sorry.)